components/integrations_marketplace/marketplace_item/marketplace_item.jsx

```diff
@@ -0,0 +1,186 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import {FormattedMessage} from 'react-intl';
+import {Link} from 'react-router-dom';
+
+import {MarketplacePluginStatus} from 'mattermost-redux/constants/plugins';
+
+import LoadingWrapper from 'components/widgets/loading/loading_wrapper.jsx';
+import ConfirmModal from 'components/confirm_modal.jsx';
+import PluginIcon from 'components/widgets/icons/plugin_icon.jsx';
+
+import {localizeMessage} from 'utils/utils';
+
+export default class MarketplaceItem extends React.Component {
+        static propTypes = {
+            id: PropTypes.string.isRequired,
+            name: PropTypes.string.isRequired,
+            description: PropTypes.string.isRequired,
+            version: PropTypes.string.isRequired,
+            downloadUrl: PropTypes.string,
+            homepageUrl: PropTypes.string,
+            installed: PropTypes.bool.isRequired,
+            onConfigure: PropTypes.func.isRequired,
+            onInstalled: PropTypes.func.isRequired,
+            actions: PropTypes.shape({
+                installPluginFromUrl: PropTypes.func.isRequired,
+            }).isRequired,
+        };
+
+        constructor(props) {
+            super(props);
+
+            this.state = {
+                installed: this.props.installed,
```

Comment:
doesn't look like this `installed` field is set anything, worth just using `this.props.installed` all the time?

Reply:
This is passed in to the local state because if its installed it changes it locally to update the button to `Configure`. It's being used here: https://github.com/mattermost/mattermost-webapp/blob/bde6b01e23ddd5bc8a9292991b850fb4c067eba5/components/integrations_marketplace/marketplace_item/marketplace_item.jsx#L81
Pushing a missing change I have locally that missed the last commit that sets `installed` instead.

Reply:
https://github.com/mattermost/mattermost-webapp/pull/3492/commits/8d46b8ceba0236e592ffbf70d6fda20fd1aae734